server/src/test/java/org/opensearch/common/xcontent/XContentTypeTests.java

```diff
@@ -43,71 +43,79 @@ public class XContentTypeTests extends OpenSearchTestCase {
     }
 
     public void testFromWildcardUppercase() throws Exception {
         String mediaType = "APPLICATION/*";
         XContentType expectedXContentType = XContentType.JSON;
-        assertThat(XContentType.fromMediaTypeOrFormat(mediaType), equalTo(expectedXContentType));
-        assertThat(XContentType.fromMediaTypeOrFormat(mediaType + ";"), equalTo(expectedXContentType));
+        assertThat(XContentType.fromMediaType(mediaType), equalTo(expectedXContentType));
+        assertThat(XContentType.fromMediaType(mediaType + ";"), equalTo(expectedXContentType));
     }
 
     public void testFromRubbish() throws Exception {
-        assertThat(XContentType.fromMediaTypeOrFormat(null), nullValue());
-        assertThat(XContentType.fromMediaTypeOrFormat(""), nullValue());
-        assertThat(XContentType.fromMediaTypeOrFormat("text/plain"), nullValue());
-        assertThat(XContentType.fromMediaTypeOrFormat("gobbly;goop"), nullValue());
+        assertThat(XContentType.fromMediaType(null), nullValue());
+        assertThat(XContentType.fromMediaType(""), nullValue());
+        assertThat(XContentType.fromMediaType("text/plain"), nullValue());
+        assertThat(XContentType.fromMediaType("gobbly;goop"), nullValue());
     }
 
     public void testVersionedMediaType() throws Exception {
-        assertThat(XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+json;compatible-with=7"), equalTo(XContentType.JSON));
-        assertThat(XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+yaml;compatible-with=7"), equalTo(XContentType.YAML));
-        assertThat(XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+cbor;compatible-with=7"), equalTo(XContentType.CBOR));
-        assertThat(XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+smile;compatible-with=7"), equalTo(XContentType.SMILE));
+        assertThat(XContentType.fromMediaType("application/vnd.opensearch+json;compatible-with=7"), equalTo(XContentType.JSON));
+        assertThat(XContentType.fromMediaType("application/vnd.opensearch+yaml;compatible-with=7"), equalTo(XContentType.YAML));
+        assertThat(XContentType.fromMediaType("application/vnd.opensearch+cbor;compatible-with=7"), equalTo(XContentType.CBOR));
+        assertThat(XContentType.fromMediaType("application/vnd.opensearch+smile;compatible-with=7"), equalTo(XContentType.SMILE));
 
-        assertThat(XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+json ;compatible-with=7"), equalTo(XContentType.JSON));
+        assertThat(XContentType.fromMediaType("application/vnd.opensearch+json ;compatible-with=7"), equalTo(XContentType.JSON));
         assertThat(
-            XContentType.fromMediaTypeOrFormat("application/vnd.opensearch+json ;compatible-with=7;charset=utf-8"),
+            XContentType.fromMediaType("application/vnd.opensearch+json ;compatible-with=7;charset=utf-8"),
```

Comment:
Could we add the test case(s) with uppercase for  `application/vnd.opensearch` (since we have special treatment)?

Reply:
done.. and good call out!